src/tribol/interface/mfem_tribol.cpp

```diff
@@ -18,16 +16,53 @@ void registerMfemCouplingScheme( IndexT cs_id, int mesh_id_1, int mesh_id_2, con
                                  const mfem::ParGridFunction& current_coords, std::set<int> b_attributes_1,
                                  std::set<int> b_attributes_2, ContactMode contact_mode, ContactCase contact_case,
                                  ContactMethod contact_method, ContactModel contact_model,
-                                 EnforcementMethod enforcement_method, BinningMethod binning_method )
+                                 EnforcementMethod enforcement_method, BinningMethod binning_method,
+                                 ExecutionMode exec_mode )
 {
+  // verify valid execution mode and set memory space
+  MemorySpace mem_space = MemorySpace::Host;
+#ifdef TRIBOL_USE_CUDA
+  if ( exec_mode == ExecutionMode::Cuda ) {
+    mem_space = MemorySpace::Device;
+    SLIC_ERROR_ROOT_IF( !mfem::Device::Allows( mfem::Backend::CUDA_MASK ), "CUDA execution is not enabled in MFEM." );
```

Comment:
Let's discuss when/where/how TRIBOL_USE_CUDA could be defined, for example, but CUDA is not enabled in MFEM.

Reply:
Additionally, we probably want to error out here because we are in some inconsistent build state, for lack of a better term, but let's discuss this.